src/understand-tidb/transaction_kv.md

```diff
@@ -0,0 +1,103 @@
+
+# Transaction Log Replication
+
+In TiDB the key space is splitted into different ranges or regions, each region is a raft group and its leader will be responsible for handling its key range related read/write requests. 
+
+If the transaction command processing in the transaction scheduler is successful, the generated transaction writes will be written into the raft log engine by the region leaders in `raftStore`(raftStore will be introduced in other documents in details). The work flow is like this:
+
+![raftStore](../img/raftStore.png)
+
+The writes generated by transaction commands will be sent to the raft peer message task queue first, then the raft batch system will poll each raft peer and handle these requests in the raft thread-pool. After all the raft logs are persisted on majority raft group members, they are regarded as `commit`. Then the correspond apply task be delivered to the apply worker pool to apply the actual write contents to the storage engine, after that the transaction command processing is considered successful and the callback will be invoked to response `OK` results to the RPC client.
+
+
+# Transaction Record In TiKV
+
+In TiDB a transaction is considered committed only if its primary key lock is committed succesfully (if async commit protocol is not used). The actual key and value written into storage engine is in the following format:
+
+
+| CF	| RocksDB Key |	RocksDB Value |
+| ---   | --- | --- |
+|Lock	 |user_key	 | lock_info |
+|Default |{user_key}{start_ts}	| user_value |
+|Write	 |{user_key}{commit_ts}	| write_info |
+
+After `prewrite`, the lock correspond records for the transaction will be written into the storage. Read and write conflicts on the "locked" key will need to consider if it's safe to bypass the lock or it must try to resolve the encountered locks. As `commit_ts` is part of the stored key, there could be different historical versions for it, and GC is responsible to clean up all the old versions which will not be needed any more. GC mechanism will be introduced in another document.
+
+# Transaction Recovery
+
+In TiDB the transaction coordinator(in `tidb-server`) is stateless and it will not persist any information. If the transaction coordinator fails for example the `tidb-server` crashes, the transaction context in memory will get lost, and as the coordinator is gone the normal commit processing will stop. How to recover the transaction state and make a decision if it should commit or abort? 
+
+Actually there is no special mechanism to recover the failed transctions, the recovery is done by other concurrent conflict transactions, or the conflict transactions will help decide the actual states of the failed tranactions. The lock resolve process will be triggered if current ongoing transaction encounters locks of other transactions doing reads or writes. The whole resolve process will be introduced in other documents in details.
```

Comment:
The expression "failed transctions" is a little misunderstanding. Will "undetermined-status transactions" be more proper?